studio/hooks/analytics/useLogsPreview.tsx

```diff
@@ -61,6 +61,13 @@ function useLogsPreview(
   // handle url generation for log pagination
   const getKeyLogs: SWRInfiniteKeyLoader = (_pageIndex: number, prevPageData: Logs) => {
     let queryParams
+
+    // check that SQL is ready for the logs request
+    if(!params.sql || !params.rawSql) {
```

Comment:
we can cut it down to:
```js
// cancel request if no sql provided
if (!params.sql) return null
```
only `sql` param is used in LF. also, useSWRInfinite uses `null` to cancel requests 

https://swr.vercel.app/docs/pagination#example-2-cursor-or-offset-based-paginated-api

Reply:
forgot to push up a change that changed it to null.
although I've cited the example to https://swr.vercel.app/docs/conditional-fetching#conditional in the comments